hashintel/hash
# Upgrade Type System Package version

## 🌟 What is the purpose of this PR?

This follows on from #996, it modifies the Graph and the Workspace backend to use the new version of the Type System Package which now contains Entity Types and Link Types. 

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1200211978612931/1202775849963301/f) _(internal)_

## 🔍 What does this change?

- Updates the dependencies on the type system package to use the [newest merge to main commit](https://github.com/blockprotocol/blockprotocol/commit/d33a4ecc214cfdcf97dd115442ef00cb6b3f6636) for the type system package
- Removes redundant definitions for `EntityType` and `LinkType` from the Graph
- Updates Graph to use the definitions from the Type System package
- Updates the REST API handlers to handle serialization and deserialization of the new types
- Updates Graph tests to use the new types and be compliant with new parts of the schemas
- Updates the Workspace backend to use the new types, modifying shims and GraphQL resolvers where necessary

## 📜 Does this require a change to the docs?

- This shouldn't be an external facing change to my knowledge. Although the OpenAPI spec has been regenerated

## ⚠️ Known issues

- Some comment threads have been opened on specific areas (see PR)
- The OpenAPI generated types are less specific than the ones from the Type System Package which means we have to do some explicit "as" assertions to get around TypeScript errors, TODOs and warnings have been added.

## 🛡 What tests cover this?

- Graph unit tests
- Graph integration tests
- Graph OpenAPI tests
- Workspace integration tests
- Workspace manual tests on the `type-editor` route

## ❓ How to test this?

1. Checkout the branch / view the deployment
2. Follow the Graph README for running the full suite of tests
3. Run `yarn external-services up --build` to rebuild the docker image
4. Follow the HASH readme and run the workspace tests. Expect many failures due to the state of `dev/graph` at the moment, model class tests and tests related to types and entities should pass.
5. Run `yarn dev`, login, and go to `/type-editor` and ensure the create property type button works.
